OpenRA.Mods.Common/Traits/World/ControlGroups.cs

```diff
@@ -0,0 +1,140 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2021 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public class ControlGroupsInfo : TraitInfo
+	{
+		public readonly int GroupsCount = 10;
```

Comment:
What if we instead defined
```csharp
public readonly string[] Groups = { "1", "2", "3", "4", "5", "6", "7", "8", "9", "0" };`
```
?

`WithTextControlGroupDecoration` and the hotkey config could then use these labels directly, and mods could give them letters or symbols if they wanted.

Reply:
Interesting idea, I will give it a go :+1:

Reply:
Worked out nicely and `GroupsCount` became unnecessary.